src/element/textElement.ts

```diff
@@ -419,6 +419,18 @@ export const getTextHeight = (
   return getLineHeightInPx(fontSize, lineHeight) * lineCount;
 };
 
+const splitTextByToken = (text: string, token: string) => {
+  const words = text.split("-");
+  if (words.length > 1) {
+    words.forEach((word, index) => {
+      if (index !== words.length - 1) {
+        words[index] = word += "-";
```

Comment:
why are we keeping the `-`? E.g. `she-wolf` will be turned to `she- wolf`. Is it to keep the `-` to contribute to width calculation? Will not the extra spaces be similarly problematic?

Reply:
For splitting and joining we use the space delimeter.
There is no extra space which gets added since we are splitting and joining via space delimeter at the end. so `she-wolf` -> ["she-", "wolf"]

Reply:
To simplify there could be multiple delimeters for splitting but uiltimately we need the list of words on which we run the wrap algo hence we split by that token (eg hyphen) -> join by space -> split by space so `space` is the final delimeter which we were and are using to split and join so tomorrow another token if we have to introduce -> we split by that token -> join by space -> finally split by space to have the right list of words

Reply:
why not just have a single `parseTokens` function for both?

```ts
const parseTokens = (text: string) => {
  const tokens = [];
  let buffer = '';

  for (let i = 0; i < text.length; i++) {
    const char = text[i];

    if (char === ' ' || char === '-') {
      if (char === '-') {
        buffer += char;
      }
      if (buffer.length > 0) {
        tokens.push(buffer);
        buffer = '';
      }
    } else {
      buffer += char;
    }
  }

  if (buffer.length > 0) {
    tokens.push(buffer);
  }

  return tokens;
}
```

Reply:
to make it clear

Reply:
Isn't this making it more complex readability wise?
I can instead do it the same way as is now but instead rename it to `parseTokens` and move the splitting of space there as well then I think there won't be any confusion. Does that help ?

Reply:
I have made the changes, let me know if its good now

Reply:
I wouldn't say it's less readable because (especially if commented). And it'll be way faster. Also, the algo you're using isn't quite correct rn: it splits `wolf-` into `["wolf-", ""]`.

Reply:
> Sadly thats how split function behaves when the last char is the split delimeter.

It may not cause issues now, but may in the future. Best to make the parser behave correctly (and which will also be faster) from the start and avoid potential issues later.

Reply:
To summarise as discussed in chat - here is a quick perf testing - looks like my algo is faster than the for loop (Small is mine, long is for loop)
![image](https://user-images.githubusercontent.com/11256141/233266115-e87791b0-812e-4e70-a297-34975fc1bbcf.png)

Tried with some random large testcase (prefilled in CSB),
If anyone wants to test it out [https://6w38zw.csb.app/](https://6w38zw.csb.app/)

As per the empty string getting added - right now it won't matter as adding empty string will not make any difference but I agree that we should remove the empty token (without compromising the readability) so algo is more accurate.